infra/domain/main.tf

```diff
@@ -4,39 +4,25 @@ data aws_route53_zone selected {
   name = "${var.DSS_ZONE_NAME}"
 }
 
-resource "aws_acm_certificate" "cert" {
-  domain_name               = "${var.DSS_CERTIFICATE_DOMAIN}"
-  subject_alternative_names = "${compact(split(" ", var.DSS_CERTIFICATE_ADDITIONAL_NAMES))}"
-  validation_method         = "${var.DSS_CERTIFICATE_VALIDATION}"
+data "aws_acm_certificate" "cert" {
+  domain      = "${var.DSS_CERTIFICATE_DOMAIN}"
+  statuses    = ["ISSUED"]
+  most_recent = true
 }
 
-resource "aws_route53_record" "cert_validation" {
-  count   = "${"DNS" == var.DSS_CERTIFICATE_VALIDATION ? 1 : 0}"
-  name    = "${aws_acm_certificate.cert.domain_validation_options.0.resource_record_name}"
-  type    = "${aws_acm_certificate.cert.domain_validation_options.0.resource_record_type}"
-  zone_id = "${data.aws_route53_zone.selected.zone_id}"
-  records = ["${aws_acm_certificate.cert.domain_validation_options.0.resource_record_value}"]
-  ttl     = 300
-}
-
-resource "aws_acm_certificate_validation" "cert_dns" {
-  count   = "${"DNS" == var.DSS_CERTIFICATE_VALIDATION ? 1 : 0}"
-  certificate_arn         = "${aws_acm_certificate.cert.arn}"
-  validation_record_fqdns = ["${aws_route53_record.cert_validation.fqdn}"]
-}
+resource "aws_api_gateway_domain_name" "dss" {
+  domain_name               = "${var.API_DOMAIN_NAME}"
+  regional_certificate_arn  = "${data.aws_acm_certificate.cert.arn}"
 
-resource "aws_acm_certificate_validation" "cert_email" {
-  count   = "${"EMAIL" == var.DSS_CERTIFICATE_VALIDATION ? 1 : 0}"
-  certificate_arn = "${aws_acm_certificate.cert.arn}"
+  endpoint_configuration {
+    types = ["REGIONAL"]
+  }
 }
 
-# TODO: Configure regional domain name with Terraform
-#   Terraform does not currently support regional API Gateway endpoints:
-#   https://github.com/terraform-providers/terraform-provider-aws/issues/2195
-resource null_resource dss_domain {
-  count  = "${length(var.DSS_CERTIFICATE_VALIDATION ) > 0 ? 1 : 0}"
-  provisioner "local-exec" {
-    command = "./create_regional_domain_name.py --domain-name ${var.API_DOMAIN_NAME} --certificate-arn ${aws_acm_certificate.cert.arn} --zone-id ${data.aws_route53_zone.selected.zone_id}"
-  }
-  depends_on = ["aws_acm_certificate_validation.cert_dns", "aws_acm_certificate_validation.cert_email"]
+resource "aws_route53_record" "dss" {
+  zone_id = "${data.aws_route53_zone.selected.zone_id}"
```

Comment:
Should this be `regional_zone_id`?

Reply:
nope, there is no such parameter for the `aws_route53_record` resource: https://www.terraform.io/docs/providers/aws/r/route53_record.html